Marlin/src/HAL/STM32F1/inc/Conditionals_LCD.h

```diff
@@ -25,3 +25,13 @@
   //#warning "SD_CHECK_AND_RETRY isn't needed with USE_USB_COMPOSITE."
   #undef SD_CHECK_AND_RETRY
 #endif
+
+#if HAS_GRAPHICAL_TFT
+  #error "Sorry! TFT displays are not available for HAL/STM32F1."
+#endif
```

Comment:
This is incompatible with `#define TFT_320x240` in the updated `Mks/Robin/Configuration.h`.

Reply:
It should be incompatible - IO is implemented for STM32 only.
STM32F1 should use `FSMC_GRAPHICAL_TFT` option which enabled `HAS_GRAPHICAL_LCD`

Reply:
Technically `TFT_320x240` and `TFT_480x320` can be redefined as
```
#define FSMC_GRAPHICAL_TFT
#define LCD_PIXEL_WIDTH 320 (480)
#define LCD_PIXEL_HEIGHT 240 (320)
#define LCD_PIXEL_OFFSET_X <something>
#define LCD_PIXEL_OFFSET_Y <something>
#define FSMC_UPSCALE 2 (3)
```
but it needs some testing and may be tweaking.

Do you want to unify there settings between old and new implementations?

Reply:
> It should be incompatible

Then I am confused about why  `TFT_320x240` was added and enabled in the Mks/Robin configuration:

https://github.com/MarlinFirmware/Configurations/pull/169/files#diff-ab7fde72990dd0d31b19bccb6921b58aR2150

It would be good to get these LCD settings fixed and settled, fix the now-broken configuration, and simplify these settings as much as possible so users don't need to concern themselves with 5 or 6 settings just to enable an LCD.

Reply:
Updated configuration is for new build environment `[env:mks_robin_stm32]` that uses HAL STM32 (CubeMX)
Its fully functional but not backward compatible with `[env:mks_robin]` (libmaple)

I can tweak `HAL\STM32F1\inc\Conditionals_LCD.h` to transform `TFT_320x240` and `TFT_480x320` to settings for upscaled 128x64 UI implementation.
This will cause confusion as well - same settings will provide different UI depending on which build environment (HAL) was used.

What solution will be lesser evil - different configuration settings to enable different type of UI or single configuration parameter that enables one of two UIs?

Reply:
Since the change to the config breaks the `mks_robin-tests` compilation, that test has been disabled. In its place I've added the `mks_robin_35-tests` file, which should compile successfully. If we can modify `mks_robin-tests` so it enables a different LCD then that test can be enabled again. Or, if `env:mks_robin` is going to be retired anyway then we can remove its test altogether.

Reply:
The `_35` suffix is a bit misleading
MKS Robin TFT3.5 has 480x320 resolution. It is used in MKS Robin Nano 1.x
Original configuration is for 320x240 screens - MKS Robin TFT versions 1.1 and 2.0 with are 2.8 and 3.2 inches

Reply:
The `mks_robin_35` test is not passing with the latest Mks/Robin configurations either. This PR and the Configuration for Mks/Robin will need to be fixed.

It appeared that a few occurrences of `TOUCH_BUTTONS` that were intended to be replaced by `TOUCH_SCREEN` got missed. I patched that but maybe these changes in 829b0abd8f should actually be `HAS_TOUCH_XPT2046` instead of `TOUCH_SCREEN`?

The problem is one of missing dependencies. I can push a couple of changes that get a little closer to a successful compile, but I don't know the ultimate resolution.

Reply:
I'm just about caught up with myself….

Since `HAS_TOUCH_XPT2046` and `TOUCH_SCREEN` are not (necessarily) both enabled, but they both need `TOUCH_*_PIN` defines, it seems that they need to both be checked for in pins files, and perhaps elsewhere.

Compilation successful.